### PR TITLE
[MRG] added --name to merge

### LINKED ISSUE #1213: sourmash sig merge needs --name
`sourmash sig merge` should support `--name` to rename the output signature.

feature suggested by @olgabot as per below.

To get started on this,
* add a `--name` parameter to argument parsing for `sig merge` in `src/sourmash/cli/sig/merge.py`
* in `src/sourmash/sig/__main__.py`, function `merge(...)`, adjust the `SourmashSignature` object creation to use `name=args.name`
* add a test in `tests/test_test_cmd_signature.py`, perhaps starting from a renamed copy of `test_sig_merge_1`, that uses `--name` on the command line and checks the `.name` attribute on the output signature

### PR DESCRIPTION
**Pull request by hehouts**
`sourmash sig merge` now supports optional `--name` to set the name of the merged signature. 

fixes https://github.com/dib-lab/sourmash/issues/1213

### CODE REVIEW

**Comment by ctb** [comment]
hi @hehouts please be sure to add your ORCID here - thanks!

**Comment by hehouts** [comment]
> hey @ctb .
My OrcID is 0000-0002-7954-4793, is there a way to link it to the pull request or something?

**Comment by ctb** [comment]
On Fri, Apr 23, 2021 at 05:48:32PM -0700, Hannah Eve Houts wrote:
> > hey @ctb .
> My OrcID is 0000-0002-7954-4793, is there a way to link it to the pull request or something?

you just did!

(no, there's no official way to link it :)

**Comment by hehouts** [comment]
@ctb @bluegenes: please review

File: tests/test_cmd_signature.py [hunk 1]
```diff
@@ -90,6 +90,25 @@ def test_sig_merge_1_multisig(c):
     assert actual_merge_sig.minhash == test_merge_sig.minhash
 
 
+@utils.in_tempdir
+def test_sig_merge_1_name(c):
+    # check name arg
+    sig2 = utils.get_test_data('2.fa.sig')
+    sig63 = utils.get_test_data('63.fa.sig')
+
+    assignedSigName = 'SIG_NAME'
```
Comment (ctb) [inline tests/test_cmd_signature.py:99]: for future reference, we _typically_ use non-CamelCase variables for things that aren't class names. So it's `MinHash` and `SourmashSignature` but `load_signature`. In this case, I would suggest `assigned_sig_name`. As this is a test, however, there's no need to change it, this is just an FYI!

**Review by ctb — APPROVED** [review]
LGTM! Do you have the "merge and squash" button available to you, @hehouts? If so, please do the honors :)

**Pull request merged**